### ISSUE TITLE
citation_to_bibkey API function

### ISSUE THREAD

**Issue by xrotwang** [glottolog/pyglottolog#21]
Implement a function 
```python
citation_to_bibkey("Safar & Webster 2014:9", bibfiles, iso, languoids) = "h:SafarWebster:Endangered-Sign"
```

The `languoids` argument would be a selection of languoids to consider. For
the use case I am thinking of, building the full index is probably
overkill. I am thinking just read the bibfiles, specified, select on the
languoids argument, and search linearly through that selection. One could
also consider an "searchin" argument where you give a string "Quechuan",
iso "eng" or languoid "blaa1234" and the search is through *all* languoids
in the subtree identified by the string.

**Comment by xrotwang** [comment]
See https://github.com/glottolog/glottolog/pull/472

**Comment by xrotwang** [comment]
@d97hah The only problem I have with that kind of functionality is the following: How to you treat ambiguity? Do you just return the first hit? Is not finding any an error? `citation_to_bibkey` sounds as if there should be exactly one hit - because a citation is supposed to identify a source unambiguously. Maybe we should start with that?

Alternatively we could go for `iter_bibkeys` and just `yield` hits, so the caller can decide if the first is good enough?

**Comment by d97hah** [comment]
Yes, return iterator, users busines to sort out 0, 1 or more hits. rename
function to _bibkeys accordingly

Pada tanggal Sel, 18 Feb 2020 pukul 14.00 Robert Forkel <
notifications@github.com> menulis:
